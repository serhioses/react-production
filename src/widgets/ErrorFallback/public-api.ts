export { ErrorFallback } from 'widgets/ErrorFallback/ui/ErrorFallback';
