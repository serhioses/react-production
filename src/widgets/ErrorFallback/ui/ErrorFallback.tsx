import { FallbackProps } from 'react-error-boundary';
import { useTranslation } from 'react-i18next';

import { Alert, AlertDescription, AlertIcon, AlertTitle } from '@chakra-ui/react';

export function ErrorFallback({ error }: FallbackProps) {
  const { t } = useTranslation();

  return (
    <Alert
      status="error"
      sx={{ flexDirection: 'column', alignItems: 'center', justifyContent: 'center' }}
    >
      <AlertIcon />
      <AlertTitle>{t('errorFallbackTitle')}</AlertTitle>
      <AlertDescription>{error.message}</AlertDescription>
    </Alert>
  );
}
