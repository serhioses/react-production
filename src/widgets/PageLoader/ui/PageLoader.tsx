import { Center, Spinner, SpinnerProps } from '@chakra-ui/react';

export function PageLoader(props: SpinnerProps) {
  return (
    <Center sx={{ position: 'absolute', inset: 0 }}>
      <Spinner size="xl" {...props} />
    </Center>
  );
}
