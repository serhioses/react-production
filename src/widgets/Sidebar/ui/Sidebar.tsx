import { useState } from 'react';

import { Box, Button } from '@chakra-ui/react';

import GradeIcon from 'shared/assets/icons/grade.svg';

export function Sidebar() {
  const [collapsed, setCollapsed] = useState(false);

  function toggleCollapsed(): void {
    setCollapsed((c) => !c);
  }

  return (
    <Box
      sx={{
        width: collapsed ? '80px' : '300px',
        height: 'calc(100dvh - 50px)',
        background: 'black',
        position: 'relative',
        transition: '0.1s ease-in',
      }}
      data-testid="sidebar"
    >
      <Button onClick={toggleCollapsed} data-testid="sidebar-toggle">
        <GradeIcon />
      </Button>
    </Box>
  );
}
