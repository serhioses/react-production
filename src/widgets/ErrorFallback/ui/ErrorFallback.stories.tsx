import type { Meta, StoryObj } from '@storybook/react';

import { ErrorFallback } from 'widgets/ErrorFallback/ui/ErrorFallback';

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
  title: 'Widgets/ErrorFallback',
  component: ErrorFallback,
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
    layout: 'centered',
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
} satisfies Meta<typeof ErrorFallback>;

export default meta;

type Story = StoryObj<typeof meta>;

const Template: Story = {
  render: (args) => {
    return <ErrorFallback {...args} />;
  },
  args: {
    error: { message: 'msg' },
    resetErrorBoundary: () => null,
  },
};

export const Basic = {
  ...Template,
};
