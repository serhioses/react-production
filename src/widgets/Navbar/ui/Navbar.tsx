import { useTranslation } from 'react-i18next';
import { useSelector } from 'react-redux';
import { Link as RouterLink } from 'react-router-dom';

import { MoonIcon, SunIcon } from '@chakra-ui/icons';
import {
  Box,
  IconButton,
  Flex,
  Link,
  SystemStyleObject,
  useColorMode,
  Button,
  HStack,
  StackDivider,
} from '@chakra-ui/react';

import { ERoutes } from 'shared/enums/routes';
import { useLazyModal } from 'shared/hooks/useLazyModal';

import { selectUserAuthData, userActions } from 'entities/User/public-api';

import { LoginModalLazy } from 'features/AuthByUsername/public-api';

import { useAppDispatch } from 'app/providers/StoreProvider/config/hooks';

type TNavbarProps = {
  sx?: SystemStyleObject;
};

export function Navbar({ sx }: TNavbarProps) {
  const { toggleColorMode, colorMode } = useColorMode();
  const { t, i18n } = useTranslation();
  const { ModalComponent: LoginModal, isOpen, onOpen, onClose } = useLazyModal(LoginModalLazy);
  const authData = useSelector(selectUserAuthData);
  const dispatch = useAppDispatch();

  function changeLang(lng: string): void {
    if (lng !== i18n.language) {
      i18n.changeLanguage(lng);
    }
  }

  function logOut(): void {
    dispatch(userActions.resetAuthData());
  }

  return (
    <Box
      sx={{
        display: 'flex',
        alignItems: 'center',
        width: '100%',
        height: 50,
        padding: 5,
        backgroundColor: colorMode === 'light' ? 'gray.500' : 'white',
        color: colorMode === 'light' ? 'white' : 'black',
        boxShadow: 'md',
        ...sx,
      }}
      data-testid="navbar"
    >
      <Flex sx={{ marginInline: 'auto 24px', gap: 4 }}>
        <Link as={RouterLink} to={ERoutes.Main}>
          {t('home')}
        </Link>
        <Link as={RouterLink} to={ERoutes.About}>
          {t('about')}
        </Link>
      </Flex>
      <IconButton
        variant="ghost"
        colorScheme={colorMode === 'light' ? 'gray' : ''}
        onClick={toggleColorMode}
        icon={colorMode === 'light' ? <MoonIcon /> : <SunIcon />}
        aria-label={t('toggleTheme')}
      />
      <HStack divider={<StackDivider />} marginInlineEnd={8}>
        <Button
          size="sm"
          variant="link"
          color="inherit"
          sx={{ opacity: i18n.language !== 'ru' && i18n.language !== 'ru-RU' ? 1 : 0.75 }}
          onClick={() => changeLang('en')}
        >
          EN
        </Button>
        <Button
          size="sm"
          variant="link"
          color="inherit"
          sx={{ opacity: i18n.language === 'ru' || i18n.language === 'ru-RU' ? 1 : 0.75 }}
          onClick={() => changeLang('ru')}
        >
          RU
        </Button>
      </HStack>
      {authData ? (
        <Button
          onClick={logOut}
          variant="link"
          color="inherit"
          sx={{ textDecoration: 'underline' }}
        >
          {t('logout')}
        </Button>
      ) : (
        <>
          <Button
            onClick={onOpen}
            variant="link"
            color="inherit"
            sx={{ textDecoration: 'underline' }}
          >
            {t('login')}
          </Button>
          <LoginModal isOpen={isOpen} onClose={onClose} />
        </>
      )}
    </Box>
  );
}
