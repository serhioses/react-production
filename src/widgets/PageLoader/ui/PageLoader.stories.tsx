import type { Meta, StoryObj } from '@storybook/react';

import { PageLoader } from 'widgets/PageLoader/ui/PageLoader';

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
  title: 'Widgets/PageLoader',
  component: PageLoader,
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
    layout: 'centered',
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes
  argTypes: {
    color: { control: 'text' },
    emptyColor: { control: 'text' },
    label: { control: 'text' },
    size: {
      options: ['xs', 'sm', 'md', 'lg', 'xl'],
      control: 'select',
    },
    speed: { control: 'text' },
    thickness: { control: 'text' },
    variant: { control: 'text' },
  },
  args: {
    size: 'sm',
  },
} satisfies Meta<typeof PageLoader>;

export default meta;

type Story = StoryObj<typeof meta>;

const Template: Story = {
  render: (args) => {
    return <PageLoader {...args} />;
  },
};

export const Basic = {
  ...Template,
};
