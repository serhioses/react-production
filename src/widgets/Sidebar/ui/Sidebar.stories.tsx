import type { Meta, StoryObj } from '@storybook/react';

import { Sidebar } from 'widgets/Sidebar/ui/Sidebar';

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export

const meta = {
  title: 'Widgets/Sidebar',
  component: Sidebar,
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
} satisfies Meta<typeof Sidebar>;

export default meta;

type Story = StoryObj<typeof meta>;

const Template: Story = {
  render: (args) => {
    return <Sidebar {...args} />;
  },
};

export const Basic = {
  ...Template,
};
