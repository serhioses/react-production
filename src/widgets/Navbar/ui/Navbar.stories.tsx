import type { Meta, StoryObj } from '@storybook/react';

import { Navbar } from 'widgets/Navbar/ui/Navbar';

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export

const meta = {
  title: 'Widgets/Navbar',
  component: Navbar,
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
} satisfies Meta<typeof Navbar>;

export default meta;

type Story = StoryObj<typeof meta>;

const Template: Story = {
  render: (args) => {
    return <Navbar {...args} />;
  },
};

export const Basic = {
  ...Template,
};
