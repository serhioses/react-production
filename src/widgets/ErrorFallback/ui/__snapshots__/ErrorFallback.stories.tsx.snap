// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widgets/ErrorFallback Basic smoke-test 1`] = `
<div data-status="error"
     role="alert"
     class="chakra-alert css-1lxprk3"
>
  <span data-status="error"
        class="chakra-alert__icon css-14ogjxt"
  >
    <svg viewbox="0 0 24 24"
         focusable="false"
         class="chakra-icon css-qk6lof"
    >
      <path fill="currentColor"
            d="M11.983,0a12.206,12.206,0,0,0-8.51,3.653A11.8,11.8,0,0,0,0,12.207,11.779,11.779,0,0,0,11.8,24h.214A12.111,12.111,0,0,0,24,11.791h0A11.766,11.766,0,0,0,11.983,0ZM10.5,16.542a1.476,1.476,0,0,1,1.449-1.53h.027a1.527,1.527,0,0,1,1.523,1.47,1.475,1.475,0,0,1-1.449,1.53h-.027A1.529,1.529,0,0,1,10.5,16.542ZM11,12.5v-6a1,1,0,0,1,2,0v6a1,1,0,1,1-2,0Z"
      >
      </path>
    </svg>
  </span>
  <div data-status="error"
       class="chakra-alert__title css-tidvy5"
  >
    Something went wrong...
  </div>
  <div data-status="error"
       class="chakra-alert__desc css-zzks76"
  >
    msg
  </div>
</div>
<span id="__chakra_env"
      hidden
>
</span>
`;
